RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -253,52 +259,83 @@ void EGExtraInfoModifierFromDB::setEvent(const edm::Event& evt) {
     get_product(evt, imap->second.second, pho_int_vmaps);
   }
   
-  if (autoDetectBunchSpacing_) {
-      edm::Handle<unsigned int> bunchSpacingH;
-      evt.getByToken(bunchSpacingToken_,bunchSpacingH);
-      bunchspacing_ = *bunchSpacingH;
-  }
-
   edm::Handle<double> rhoH;
   evt.getByToken(rhoToken_, rhoH);
   rhoValue_ = *rhoH;
-  
-  evt.getByToken(vtxToken_, vtxH_);
-  nVtx_ = vtxH_->size();
+ 
+  evt.getByToken( ecalRecHitEBToken_, ecalRecHitsEB_ );
+  evt.getByToken( ecalRecHitEEToken_, ecalRecHitsEE_ );
+
 }
 
 void EGExtraInfoModifierFromDB::setEventContent(const edm::EventSetup& evs) {
 
   iSetup_ = &evs;
 
   edm::ESHandle<GBRForestD> forestDEH;
-  edm::ESHandle<GBRForest> forestEH;
+  GBRForestD* forest;
+  
+  const std::vector<std::string> ph_condnames_ecalonly_mean  = ph_conf.condnames_ecalonly_mean;
+  const std::vector<std::string> ph_condnames_ecalonly_sigma = ph_conf.condnames_ecalonly_sigma;
 
-  const std::vector<std::string> ph_condnames_mean  = (bunchspacing_ == 25) ? ph_conf.condnames_mean_25ns  : ph_conf.condnames_mean_50ns;
-  const std::vector<std::string> ph_condnames_sigma = (bunchspacing_ == 25) ? ph_conf.condnames_sigma_25ns : ph_conf.condnames_sigma_50ns;
+  if (useLocalFile_)
+    pointerLocalFile_ = TFile::Open(addressLocalFile_.fullPath().c_str(), "READ");
```

Comment:
file open and close every event is another no-go.
Please move to the constructor or something that's not executed every event.